core/src/main/java/io/keen/client/java/KeenClient.java

```diff
@@ -1362,8 +1364,9 @@ private void validateEventValue(Object value, int depth) {
      * @throws IOException If there was an error communicating with the server.
      */
     private String publish(KeenProject project, String eventCollection,
-                           Map<String, Object> event) throws IOException {
+                           Map<String, Object> event) throws IOException, URISyntaxException {
         // just using basic JDK HTTP library
+        eventCollection = new URI(null, null, eventCollection,null).getRawPath();
```

Comment:
how URISyntaxException is handled in places where this method is called?

Reply:
I would assume @FilipShark has investigated in what exact circumstances the exception can be throws, right? :trollface: